src/main/java/namor/template/calculator/RPNcalculator.java

```diff
@@ -35,33 +39,33 @@ private void doArithmeticOperation(String arithmeticOperator, ArrayStack numbers
         }
     }
 
-    private final int addition(ArrayStack numbers) {
-        final int secondNumber = numbers.pop();
-        final int firstNumber = numbers.pop();
+    private final double addition(Stack numbers) {
+        final double secondNumber = (Double) numbers.pop();
+        final double firstNumber = (Double) numbers.pop();
 
         return firstNumber + secondNumber;
     }
 
-    private final int subtraction(ArrayStack numbers) {
-        final int secondNumber = numbers.pop();
-        final int firstNumber = numbers.pop();
+    private final double subtraction(Stack numbers) {
+        final double secondNumber = (Double) numbers.pop();
+        final double firstNumber = (Double) numbers.pop();
 
         return firstNumber - secondNumber;
     }
 
-    private final int multiplication(ArrayStack numbers) {
-        final int secondNumber = numbers.pop();
-        final int firstNumber = numbers.pop();
+    private final double multiplication(Stack numbers) {
+        final double secondNumber = (Double) numbers.pop();
+        final double firstNumber = (Double) numbers.pop();
 
         return firstNumber * secondNumber;
     }
 
-    private final int division(ArrayStack numbers) {
-        final int secondNumber = numbers.pop();
-        final int firstNumber = numbers.pop();
+    private final double division(Stack numbers) {
+        final double secondNumber = (Double) numbers.pop();
+        final double firstNumber = (Double) numbers.pop();
 
         return firstNumber / secondNumber;
     }
 
```

Comment:
In good practices of OOP you call the methods as a verb. So not addition, but add. And change names of all the methods to verb